esphome/components/esp_ldo/esp_ldo.cpp

```diff
@@ -0,0 +1,34 @@
+#ifdef USE_ESP32_VARIANT_ESP32P4
+#include "esp_ldo.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+
+namespace esphome {
+namespace esp_ldo {
+
+static const char *const TAG = "esp_ldo";
+void EspLdo::setup() {
+  esp_ldo_channel_config_t config{};
+  config.chan_id = this->channel_;
+  config.voltage_mv = (int) (this->voltage_ * 1000.0f);
+  config.flags.adjustable = this->adjustable_;
+  auto err = esp_ldo_acquire_channel(&config, &this->handle_);
+  if (err != ESP_OK) {
+    auto msg = str_sprintf("Failed to acquire LDO channel %d with voltage %fV", this->channel_, this->voltage_);
+    this->mark_failed(msg.c_str());
```

Comment:
Isn't msg destroyed later? Component will then have dangling pointer to error message.


```suggestion
    static auto msg = str_sprintf("Failed to acquire LDO channel %d with voltage %fV", this->channel_, this->voltage_);
    this->mark_failed(msg.c_str());
```


I'm not sure if this will also work properly. Ideally there should be static buffer.

Reply:
No, it's fine. The string gets passed to `status_set_error()` which assigns it to a `std::string` thus copying the content before return from the call.